clang/tools/driver/cc1depscan_main.cpp

```diff
@@ -831,65 +817,52 @@ int cc1depscand_main(ArrayRef<const char *> Argv, const char *Argv0,
   openAndReplaceFD(1, LogOutPath);
   openAndReplaceFD(2, LogErrPath);
 
-  bool ShouldKeepOutputs = true;
-  auto DropOutputs = llvm::make_scope_exit([&]() {
-    if (ShouldKeepOutputs)
-      return;
-    ::unlink(LogOutPath.c_str());
-    ::unlink(LogErrPath.c_str());
-  });
+  Server.start(/*Exclusive*/ false);
+  return Server.listen();
+}
+
+void ScanServer::start(bool Exclusive) {
+  // Check the pidfile.
+  SmallString<128> PidPath;
+  (BasePath + ".pid").toVector(PidPath);
 
   int PidFD;
   [&]() {
     if (std::error_code EC = llvm::sys::fs::openFile(
             PidPath, PidFD, llvm::sys::fs::CD_OpenAlways,
             llvm::sys::fs::FA_Write, llvm::sys::fs::OF_None))
-      llvm::report_fatal_error("clang -cc1depscand: cannot open pidfile");
+      reportError("cannot open pidfile");
 
     // Try to lock; failure means there's another daemon running.
-    if (::flock(PidFD, LOCK_EX | LOCK_NB))
+    if (::flock(PidFD, LOCK_EX | LOCK_NB)) {
+      if (Exclusive)
+        reportError("another daemon using the base path");
```

Comment:
I think this is not an error.

Two clang can try to spawn together, and one will succeed, and the other hit this error but it can connect to the same daemon.

Reply:
That's why there's the `Exclusive` check, when clang auto-spawns this is not an error; it is an error only when you try to launch explicitly with `-serve` or `-execute`  modes.

Reply:
Ah, I missed that. 👍